util/src/servers.rs

```diff
@@ -0,0 +1,37 @@
+use glob::glob;
+use std::env;
+
+use anyhow::{Context, Result};
+
+pub fn settings_for(component: &str) -> Result<config::Config> {
+    Ok(config::Config::default()
+        .merge(
+            glob(
+                &get_conf_dir(&format!(
+                    "{}_CONF_DIR",
+                    component.to_string().replace("-", "_").to_uppercase()
+                ))
+                .unwrap_or_else(|| format!("/etc/fdo/{}/conf.d/*.yml", component)),
+            )?
+            .map(|path| config::File::from(path.unwrap()))
+            .collect::<Vec<_>>(),
+        )
+        .context("Loading configuration files from conf.d")?
+        .merge(config::File::with_name(component).required(false))
+        .context("Loading configuration file")?
+        .merge(config::Environment::with_prefix(
+            &component.to_string().replace("-", "_"),
+        ))
+        .context("Loading configuration from environment variables")?
+        .clone())
+}
```

Comment:
Should we have somewhere in /usr for hard RHEL defaults?

Should env vars take precedence over /etc as I'm guessing they may be set by a container runtime? Or is it check etc first then CWD then env aka the last read is the precedence?

Reply:
> I wonder if we should do:

I like it, I can change it to that

> I don't think I really like to automatically grab whatever's in the cwd, that was mostly there because I was just starting out with the whole config framework etc.

Perfect I can drop it :) :) :)

> Also, note that this is not /etc/fdo/{0}/conf.d/*. That looks kind of silly to me.

yeah, works for me with `{0}.conf.d/` as there are more services sharing `/etc/fdo`

I'll fix this up tomorrow

Reply:
@puiterwijk ptal - I've dropped the local file and config from env keys as well